platforms/jvm/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoReportBase.java

```diff
@@ -57,6 +60,7 @@ public abstract class JacocoReportBase extends JacocoBase {
     private final ConfigurableFileCollection classDirectories = getProject().files();
     private final ConfigurableFileCollection additionalClassDirs = getProject().files();
     private final ConfigurableFileCollection additionalSourceDirs = getProject().files();
+    private final Property<String> sourceEncoding = getProject().getObjects().property(String.class).value(Charset.defaultCharset().name());
```

Comment:
❗ The convention for a task property should be set by the Plugin after it creates that task.

Reply:
And in this case there should imho not be a default anyway and just not given anything to the Ant task if the provider is not set.